DatadogRUM/Tests/RUMMonitor/MonitorTests.swift

```diff
@@ -44,6 +44,26 @@ class MonitorTests: XCTestCase {
         XCTAssertEqual(rumContext.viewID, expectedContext.activeViewID?.toRUMDataFormat)
     }
 
+    func testWhenSessionIsNotSampled_itReturnsCurrentSessionId() throws {
+        // Given
+        let sampler = Sampler(samplingRate: 100)
+
+        // When
+        let monitor = Monitor(
+            core: core,
+            dependencies: .mockWith(core: core, sessionSampler: sampler),
+            dateProvider: DateProviderMock()
+        )
+        monitor.startView(key: "foo")
+        monitor.flush()
```

Comment:
The use of `monitor.flush()` and `PassthroughCoreMock` (as `core`) hides key problems with this design (it isn't thread-safe and is exposed to race condition). See my previous comment for insight. It would be better to add this test in [RUMMonitorTests.swift](https://github.com/DataDog/dd-sdk-ios/blob/2.5.1/DatadogCore/Tests/Datadog/RUM/RUMMonitorTests.swift) where we use a real instance of `DatadogCore` (not a single-threaded mock).